addons/mass_mailing/models/mailing.py

```diff
@@ -260,15 +260,30 @@ def _compute_statistics(self):
             self.browse(row.pop('mailing_id')).update(row)
 
     def _compute_next_departure(self):
-        cron_next_call = self.env.ref('mass_mailing.ir_cron_mass_mailing_queue').sudo().nextcall
+        cron_job = self.env.ref('mass_mailing.ir_cron_mass_mailing_queue').sudo()
         str2dt = fields.Datetime.from_string
-        cron_time = str2dt(cron_next_call)
+        cron_time = str2dt(cron_job.nextcall)
+        cron_id = cron_job.id
+
+        # The cron worker usually clean up old cron.triggers after finishing their
+        # jobs, i.e older than now() thus fetched recordset should not be to large
+        now_datetime_buffer = fields.Datetime.now() + relativedelta(minutes=1)
+        now_datetime_str = fields.Datetime.to_string(now_datetime_buffer)
+        cron_triggers = self.env['ir.cron.trigger'].sudo().search(['&', ('cron_id', '=', cron_id), ('call_at', '<', now_datetime_str)])
+        most_recent_trigger_datetime = max([str2dt(tr.call_at) for tr in cron_triggers]) if cron_triggers else False
         for mass_mailing in self:
             if mass_mailing.schedule_date:
                 schedule_date = str2dt(mass_mailing.schedule_date)
-                mass_mailing.next_departure = max(schedule_date, cron_time)
+                mass_mailing.next_departure = max(schedule_date, fields.datetime.now())
             else:
-                mass_mailing.next_departure = cron_time
+                # We checked for cron.trigger relating to the cron_id
+                # If at least one more recent than "now" exists we can reasonably
+                # expect that the cron worker will execute this mass_mailing job
+                # the next time it wakes up
+                if most_recent_trigger_datetime:
+                    mass_mailing.next_departure = max(most_recent_trigger_datetime, fields.datetime.now())
```

Comment:
I believe the search domain already guarantees that one will be greated than the other, right?

Reply:
Idem as before, we could remove the max call and just set it to most_recent_trigger_datetime as the time difference with now should be negligible.